### ISSUE TITLE
Question: installing Xournal++ under Debian Buster

### ISSUE THREAD

**Issue by JonasDralle** [xournalpp/xournalpp#959]
Hi. I'm not sure why #27 was closed.
I'd like to request a xpp build for the debian main repository.

**Comment by andreasb242** [comment]
We have currently a PPA for Ubuntu, and I have a Maintainer for Debian, but he has currently no time.
Feel free to help, all the technical Stuff is done, for the Ubuntu Build the Debian packaging was also needed.

The only thing which is missing, is a Maintainer. And no, I don't do it. No time and it's a complicated process ;-)

**Comment by JonasDralle** [comment]
Do I understand correctly, that there's already a debian package due to ubuntu support but nobody has put their name on paper as maintainer? What would a maintainer do besides redirecting all the bug reports to this repo and keeping the package up to date?

**Comment by LittleHuba** [comment]
Essentially it is the maintainers work to make sure that Xournal++ is running on the OS. That means checking dependencies and writing the build scripts for the OS (you can take the basic steps from our Wiki).
There is no need to forward any bug reports, you can just tell people to report them on our GitHub page.

**Comment by andreasb242** [comment]
>Do I understand correctly, that there's already a debian package due to ubuntu support but nobody has put their name on paper as maintainer?

Yes, absolutely.

And there is nothing to script - everything is done, Ubuntu uses Debian Packages.

The big issue: You need to be a Debian Maintainer. Read this, and you know why I didn't do this yet.
https://wiki.debian.org/DebianMaintainer

**Comment by JonasDralle** [comment]
Tried installing xournalpp by shoving the ubuntu .dep package into dpkg. Operating system is Debian Testing. Apparently `libzip5` [is missing](https://packages.debian.org/search?keywords=libzip5) in the debian repos or something.

<details>
<summary>Used commands and output</summary>

> sudo dpkg -i xournalpp_1.0.6-0\~201903161849\~ubuntu19.04.1_amd64.deb 

```
Selecting previously unselected package xournalpp.
(Reading database ... 123456 files and directories currently installed.)
Preparing to unpack .../xournalpp_1.0.6-0~201903161849~ubuntu19.04.1_amd64.deb ...
Unpacking xournalpp (1.0.6-0~201903161849~ubuntu19.04.1) ...
dpkg: dependency problems prevent configuration of xournalpp:
 xournalpp depends on libc6 (>= 2.29); however:
  Version of libc6:amd64 on system is 2.28-8.
 xournalpp depends on libportaudiocpp0; however:
  Package libportaudiocpp0 is not installed.
 xournalpp depends on libzip5 (>= 1.0); however:
  Package libzip5 is not installed.

dpkg: error processing package xournalpp (--install):
 dependency problems - leaving unconfigured
Processing triggers for mime-support (3.62) ...
Processing triggers for gnome-menus (3.31.4-3) ...
Processing triggers for desktop-file-utils (0.23-4) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Processing triggers for shared-mime-info (1.10-1) ...
Errors were encountered while processing:
 xournalpp
```

> sudo apt install -f

```
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Correcting dependencies... Done
The following packages were automatically installed and are no longer required:
  [...]
Use 'sudo apt autoremove' to remove them.
The following packages will be REMOVED:
  xournalpp
0 upgraded, 0 newly installed, 1 to remove and 0 not upgraded.
1 not fully installed or removed.
After this operation, 3,753 kB disk space will be freed.
Do you want to continue? [Y/n] y
(Reading database ... 123456 files and directories currently installed.)
Removing xournalpp (1.0.6-0~201903161849~ubuntu19.04.1) ...
Processing triggers for mime-support (3.62) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Processing triggers for gnome-menus (3.31.4-3) ...
Processing triggers for shared-mime-info (1.10-1) ...
Processing triggers for desktop-file-utils (0.23-4) ...
```

just trying it again because `*engineering intensifies*`

> sudo dpkg -i xournalpp_1.0.6-0\~201903161849\~ubuntu19.04.1_amd64.deb 

```
\~ubuntu19.04.1_amd64.deb
Selecting previously unselected package xournalpp.
(Reading database ... 418882 files and directories currently installed.)
Preparing to unpack .../xournalpp_1.0.6-0~201903161849~ubuntu19.04.1_amd64.deb ...
Unpacking xournalpp (1.0.6-0~201903161849~ubuntu19.04.1) ...
dpkg: dependency problems prevent configuration of xournalpp:
 xournalpp depends on libc6 (>= 2.29); however:
  Version of libc6:amd64 on system is 2.28-8.
 xournalpp depends on libportaudiocpp0; however:
  Package libportaudiocpp0 is not installed.
 xournalpp depends on libzip5 (>= 1.0); however:
  Package libzip5 is not installed.

dpkg: error processing package xournalpp (--install):
 dependency problems - leaving unconfigured
Processing triggers for mime-support (3.62) ...
Processing triggers for gnome-menus (3.31.4-3) ...
Processing triggers for desktop-file-utils (0.23-4) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Processing triggers for shared-mime-info (1.10-1) ...
Errors were encountered while processing:
 xournalpp
```

> sudo apt install libzip5 libportaudiocpp0

```
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Package libzip5 is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'libzip5' has no installation candidate
```
</details>

**Comment by LittleHuba** [comment]
Try `lipzip-dev` instead of `libzip5`

**Comment by JonasDralle** [comment]
That's not a debian package.
[searching for it](https://packages.debian.org/search?suite=all&section=all&arch=any&searchon=names&keywords=lipzip-dev) gave no results

**Comment by LittleHuba** [comment]
Sorry typo -_- 
It's `libzip-dev` of course.

**Comment by JonasDralle** [comment]
I have it installed but apt maybe considers `libzip-dev` and `libzip5` to be different things

```
xournalpp depends on libzip5 (>= 1.0); however:
  Package libzip5 is not installed.

dpkg: error processing package xournalpp (--install):
 dependency problems - leaving unconfigured
```

Welp, maybe one day there's someone who's interested in this project and who can ship debian packages.

**Comment by LittleHuba** [comment]
That is actually a problem that is quite easy to fix. 
Please take a look at your directory where all your libraries are stored (Should be `/usr/lib`) and search for `libzip`. We are interested in all files containing that name. Please send me a list of those files, then I can fix the build script.

**Comment by andreasb242** [comment]
The current build instructions are working on Debian Testing.
But if the .deb is liked against Ubuntu libs, it may not work with debian Libs, so I think we need a Debian Build, not a fix for the Package.

@LittleHuba if you still need the list: let me know

**Comment by LittleHuba** [comment]
This issue is just Cmake not finding the right library as it is stored using a different name. I had to fix this for LUA as well.

**Comment by andreasb242** [comment]
@LittleHuba with the current master and the current debian Testing there is absolutely no issue.
You need to install the dependencies, listed in the readme and the build is working perfect, all dependencies are there.

**Comment by LittleHuba** [comment]
@JonasDralle Please send me the list of files I requested above, so we can determine where the problem is.

**Comment by JonasDralle** [comment]
The only lib I have with `zip` in its name is `p7zip`.
[libzip-dev](https://packages.debian.org/buster/libzip-dev) is installed (double checked that apt info returns `APT-Manual-Installed: yes`)

**Comment by JonasDralle** [comment]
Sorry people i think I'll go with Xournal instead.
I know that this is a very fixable problem but for me the install process with all its github comments is sort of hitting the point of diminishing return.

Debian is tossed around as an "Ubuntuesque" and "should work" while the ubuntu .deb package does exactly nothing for me. I'm not especially interested in putting maintenance work into a package which cannot automatically update.

Consider maintaining a package in [an unofficial debian repository](https://wiki.debian.org/DebianRepository/Unofficial).

**Comment by oyren** [comment]
I needed the following Packages too build Xournalpp on Debian Buster (RC1).
```sudo apt install pkg-config libgtk2.0-dev libpoppler-dev libpoppler-glib-dev libpoppler-private-dev intltool libglade2-dev librsvg2-dev liblcms2-dev libjpeg-dev libboost-all-dev dvipng libglibmm-2.4-dev libgtk-3-dev libzip-dev portaudio19-dev libsndfile1-dev```

Then I followed the following instructions:
```git clone http://github.com/xournalpp/xournalpp
cd xournalpp
mkdir build
cd build
cmake .. # or use cmake-gui .. to customize your build more comfortable
make
make install```

**Comment by JonasDralle** [comment]
Debian Wiki: [Installing software via `make` is discouraged](https://wiki.debian.org/DontBreakDebian#A.27make_install.27_can_conflict_with_packages).

**Comment by oyren** [comment]
Sure, not the best option, i am hoping for a snap or flatpak package (#857).

**Comment by JoeKays** [comment]
It would be pretty easy to fix the debian package actually.
The missing libc6 version 2.29 is not required actually, since the older version works, libzip5 is also not required as for building from source libzip4/libzip-dev suffices. Portaudio is readily available in portaudio19-dev. So why is the debian package not simply updated?

**Comment by LittleHuba** [comment]
@andreasb242 I think this is your area of expertise 😉

**Comment by Technius** [comment]
We should fix Debian packaging as we work on #1176 and #1013.

**Comment by esternin** [comment]
To answer the implied question from @Technius, the dependencies do not appear to be set correctly for Debian 10 stable (a.k.a. buster):
```
#find /usr -name "libzip*"
/usr/lib/aarch64-linux-gnu/libzip.so.4.0
/usr/lib/aarch64-linux-gnu/libzip.so.4
/usr/lib/aarch64-linux-gnu/pkgconfig/libzip.pc
/usr/lib/aarch64-linux-gnu/libzip.so
/usr/share/doc/libzip4
/usr/share/doc/libzip-dev
/usr/share/lintian/overrides/libzip4
/usr/share/man/man3/libzip.3.gz
# apt-get install xournalpp
...
The following packages have unmet dependencies:
 xournalpp : Depends: libc6 (>= 2.29) but 2.28-10 is to be installed
             Depends: libzip5 (>= 1.0) but it is not installable
E: Unable to correct problems, you have held broken packages.
```
Any chance this can be fixed?  It is a strange thing when a program cannot be installed under the current stable version of a distribution, while claiming to be a package under that distribution.

See also in #1444
